src/status_im/ui/screens/home/views.cljs

```diff
@@ -87,15 +91,165 @@
       [react/i18n-text {:style styles/welcome-text-description
                         :key   :welcome-to-status-description}]]]))
 
-(views/defview chats-list []
-  (views/letsubs [home-items [:home-items]]
-    (if (empty? home-items)
-      [react/view styles/no-chats
-       [react/i18n-text {:style styles/no-chats-text :key :no-recent-chats}]]
-      [list/flat-list {:data      home-items
-                       :key-fn    first
-                       :render-fn (fn [home-item]
-                                    [home-list-item home-item])}])))
+(def search-input-height 56)
+
+(defn search-input
+  [search-filter {:keys [on-cancel on-focus on-change]}]
+  (let [input-is-focused? (reagent/atom false)
+        input-ref (reagent/atom nil)]
+    (fn [search-filter]
+      (let [show-cancel? (or @input-is-focused?
+                             (not-empty search-filter))]
+        [react/view {:style {:height             search-input-height
```

Comment:
Probably would make sense to separate away styles so that to aid readability.

Reply:
the style isn't reusable and since it is a view it is mostly concerned with how things should be rendered, I find it much easier to adjust styling when things are in one place, also when it comes to hunting down useless layers of components

Reply:
Got it, agree.